### ISSUE TITLE
Error and debugging.

### ISSUE THREAD

**Issue by Miracle-cr7** [spyder-ide/spyder#12862]
### What steps will reproduce the problem?

While I was coding, the error are still not being demarked after its removal,i.e., the red cross ('x') remains even after the error is rectified and the code looks erroneous. The error or syntax buffer may be the issue.

## Versions

* Spyder version: 4.1.3 
* Python version: 3.7.4
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Windows 8.1

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0           :  1.4.0 (OK)
chardet >=2.0.0                :  3.0.4 (OK)
cloudpickle >=0.5.0            :  1.4.1 (OK)
diff_match_patch >=20181111    :  20181111 (OK)
intervaltree                   :  None (OK)
IPython >=4.0                  :  7.13.0 (OK)
jedi =0.15.2                   :  0.15.2 (OK)
nbconvert >=4.0                :  5.6.1 (OK)
numpydoc >=0.6.0               :  0.9.2 (OK)
paramiko >=2.4.0               :  2.7.1 (OK)
parso =0.5.2                   :  0.5.2 (OK)
pexpect >=4.4.0                :  4.8.0 (OK)
pickleshare >=0.4              :  0.7.5 (OK)
psutil >=5.3                   :  5.7.0 (OK)
pygments >=2.0                 :  2.6.1 (OK)
pylint >=0.25                  :  2.5.2 (OK)
pyls >=0.31.9;<0.32.0          :  0.31.10 (OK)
qdarkstyle >=2.8               :  2.8.1 (OK)
qtawesome >=0.5.7              :  0.7.0 (OK)
qtconsole >=4.6.0              :  4.7.4 (OK)
qtpy >=1.5.0                   :  1.9.0 (OK)
rtree >=0.8.3                  :  0.9.4 (OK)
sphinx >=0.6.6                 :  3.0.3 (OK)
spyder_kernels >=1.9.1;<1.10.0 :  1.9.1 (OK)
watchdog                       :  None (OK)
zmq >=17                       :  18.1.1 (OK)

# Optional:
cython >=0.21                  :  0.29.17 (OK)
matplotlib >=2.0.0             :  3.1.3 (OK)
numpy >=1.7                    :  1.18.1 (OK)
pandas >=0.13.1                :  1.0.3 (OK)
scipy >=0.17.0                 :  1.4.1 (OK)
sympy >=0.7.3                  :  1.5.1 (OK)
```

**Comment by steff456** [comment]
Hi @Miracle-cr7,

Can you give us the snippet of code that doesn't show as expected? 

Right now the Spyder team is working hard to stabilize the Python Language Server, for the next release we already fixed a lot of bugs, so please let me know what error is shown in your side to see if we already fix it.

**Comment by Miracle-cr7** [comment]
Where can i share the code snippet?
Github or this mail ID

Miracle Zane

**Comment by steff456** [comment]
Github please :)

**Comment by Miracle-cr7** [comment]
![error-rectified](https://user-images.githubusercontent.com/55621586/82994640-1d592300-a020-11ea-9e8e-b7c674ccb51b.png)
Just check the red sign at line 14 which hasn't been removed even after the code executes successfully.It has been happening after the latest spyder update 4.1.3. Also the program execution is also slowed and needs to be optimised.

**Comment by steff456** [comment]
Thanks @Miracle-cr7, this issue is already fixed by PR #12711 and it will be out in Spyder's release 4.1.4
